Src/Particle/AMReX_ParticleTransformation.H

```diff
@@ -403,16 +403,19 @@ Index filterParticles (DstTile& dst, const SrcTile& src, Pred&& p,
     auto* p_mask = mask.dataPtr();
     const auto src_data = src.getConstParticleTileData();
 
-    amrex::ParallelForRNG(n,
-    [=] AMREX_GPU_DEVICE (int i, amrex::RandomEngine const& engine) noexcept
-    {
-        amrex::ignore_unused(p, p_mask, src_data, src_start, engine);
-        if constexpr (IsCallable<Pred,decltype(src_data),Index,RandomEngine>::value) {
+    if constexpr (IsCallable<Pred,decltype(src_data),Index,RandomEngine>::value) {
```

Comment:
```suggestion
    amrex::ParallelForRNG(n,
    [p, p_mask, src_data, src_start] AMREX_GPU_DEVICE (int i, amrex::RandomEngine const& engine) noexcept
    {
        amrex::ignore_unused(p, p_mask, src_data, src_start, engine);
        if constexpr (IsCallable<Pred,decltype(src_data),Index,RandomEngine>::value) {
            p_mask[i] = p(src_data, src_start+i, engine);
        } else {
            p_mask[i] = p(src_data, src_start+i);
        }
    });
```

Explicitly specifying the capture group has worked for me in the past. Putting the if constexpr outside of the kernel would result in more nvcc issues.

Reply:
Using `amrex::ignore_unused` should have the same effect as explicit capture.

Some background. A WarpX run crashes on summit in this function. So we are trying to understand what's going on. I think it's most likely a compiler bug.  CMake plus `-O2` works. CMake plus `-O3` does not. GNU Make works with `-O3`.

Reply:
The same run also works on perlmutter. And we have run it on CPU with valgrind and fsanitizer.